apps/framework-cli/src/cli/processing_coordinator.rs

```diff
@@ -0,0 +1,334 @@
+//! # Processing Coordinator Module
+//!
+//! This module provides synchronization between file watcher infrastructure processing
+//! and MCP tool requests. It ensures that MCP tools don't read partial or inconsistent
+//! state while the file watcher is applying infrastructure changes.
+//!
+//! ## Architecture
+//!
+//! The coordinator uses a generation counter pattern:
+//! - Even generations = stable state (safe to read)
+//! - Odd generations = processing in progress (wait before reading)
+//!
+//! ## Usage
+//!
+//! ```rust
+//! // In file watcher:
+//! let _guard = coordinator.begin_processing();
+//! // ... apply infrastructure changes ...
+//! // Guard drops, marking processing complete
+//!
+//! // In MCP tool:
+//! coordinator.wait_for_stable_state().await;
+//! // Now safe to read infrastructure state
+//! ```
+
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+use tokio::sync::Notify;
+
+/// Coordinates MCP requests with file watcher processing to prevent race conditions.
+///
+/// This coordinator ensures that MCP tools wait for any in-progress infrastructure
+/// changes to complete before reading state from Redis, ClickHouse, or Kafka.
+#[derive(Clone)]
+pub struct ProcessingCoordinator {
+    /// Generation counter - increments with each processing cycle
+    /// Even = stable, Odd = processing
+    generation: Arc<AtomicU64>,
+    /// Notifies waiters when processing completes
+    completion_notify: Arc<Notify>,
+}
+
+impl ProcessingCoordinator {
+    /// Create a new ProcessingCoordinator
+    pub fn new() -> Self {
+        Self {
+            generation: Arc::new(AtomicU64::new(0)),
```

Comment:
the dance for synchronization here feels very subtle
can we do a read write lock instead?